### ISSUE TITLE
Make the object_detection folder pip installable

### ISSUE THREAD

**Issue by hadim** [tensorflow/models#4887]
Please see https://github.com/tensorflow/models/issues/917#issuecomment-407552473 for history.

In short, that would allow to really use `object_detection` as a lib and an API without having to clone the whole repository.

It would make it more easy to use and everyone's code using this API more reproducible.

Now it's up to you whether you want to keep it in this repo or create a separate repo from `models`. The advantage of the later is that you can easily install whatever git commit you to want with something like that: `pip install https://github.com/tensorflow/models/archive/6c21084503b27a9ab118e1db25f79957d5ef540b.zip`.

That being said I am totally fine with regularly upload wheels to PyPi from only the `research/object_detection` folder.

The last thing for a pip installable package, you would have to compile PB files in the `setup.py` file, but that should not be too hard. I can submit a PR if you want/need.

Hope this makes sense to you.

**Comment by hadim** [comment]
I am also aware that a lot of Python files are meant to be used as runnable scripts. One option would be to make them available as scripts or entry points in the setup.py: http://python-packaging.readthedocs.io/en/latest/command-line-scripts.html

**Comment by hadim** [comment]
Here is a PR to demonstrate my point: https://github.com/tensorflow/models/pull/4889

**Comment by hadim** [comment]
@bitfort is this something you would consider?

**Comment by hadim** [comment]
ping @karmel

**Comment by yaceben** [comment]
any update on this ? thanks

**Comment by danielvdende** [comment]
This would be extremely useful to have, any update on this?

**Comment by Sri-vatsa** [comment]
Would love to help to make the folder pip installable if it is open for contributions!

**Comment by adbeda** [comment]
@hadim on issue [#62](https://github.com/conda-forge/staged-recipes/pull/8590), I attempted to bring the object_detection API to `conda-forge` compiled from a seperate github repo just for the `OO-API` but @hajapy suggested doing so from the tensorflow's official source. I noticed you tried to `pip` installable tarball but the whole effort seem open. Has there been any improvement on this direction? would you be interested in helping me add this to the conda channel?

**Comment by hadim** [comment]
Good luck with that! I don't think it will be easy as long as Google folks don't want to help to convert this repo in a python package. And I don't want to maintain a sync-fork myself... So unless a Googlers wants to help here I don't think there is much hope to get a conda forge package accepted.

**Comment by bhack** [comment]
If model is not in an internal competition with keras-cv I hope that all the reusable components (libs/utils) could be adopted directly in [keras-cv](https://github.com/keras-team/keras-cv) /cc @tanzhenyu

**Comment by bhack** [comment]
This Is just the on of the largest API surfsce exposed in model directly but more in general this topic is tracked at https://github.com/tensorflow/community/issues/223

**Comment by tanzhenyu** [comment]
Looks like this is an old thread -- I'm not sure if people are still tracking this. Keras-CV should be the place for "object detection as a lib and an API without having to clone the whole repository". If you'd like to contribute directly we'd welcome that

**Comment by bhack** [comment]
@tanzhenyu It is old but not outdated. As you can see it was just quickly re-evaulated on 9th May.
@jaeyounkim added this to `Needs triage in Object Detection on 9 May` and the new [Tensorflow object api](https://blog.tensorflow.org/2020/07/tensorflow-2-meets-object-detection-api.html) was just released (10th July)

**Comment by MarkPundurs** [comment]
Have the third-party attempts at https://pypi.org/search/?q=tensorflow+object+detection been evaluated?
